sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -272,3 +275,104 @@ def test_nested_circles():
     # The data is perfectly linearly separable in that space
     train_score = Perceptron(max_iter=5).fit(X_kpca, y).score(X_kpca, y)
     assert_equal(train_score, 1.0)
+
+
+def test_errors_and_warnings():
```

Comment:
Call this `test_kernel_conditioning` maybe?

Right now the tests don't feel very *unitary*, since you rely on the fact that the `_centerer` attribute isn't overridden when calling `fit_transform`.
Since we have a dedicated test for `check_spd_values` now, I think it'd be enough here to simply make sure that `check_spd_values` gets called, e.g. by passing a pathological X and making sure we get a warning / error from `check_spd_values`.

Reply:
If I remember well my problem was that could not create a pathological `X` here without the subsequent PR (using the approximate solver). I'll try again..

Reply:
Ok I gave up handcrafting a kernel as I do not master enough the spectrum of centered kernel matrices. So back to you idea of pathological X, I found an example leading to a small non-zero eigenvalue, I simply check that it is correctly set to zero by `check_psd_eigenvalues`.